src/hwProvider.ts

```diff
@@ -226,6 +220,19 @@ export class HWProvider {
         };
     }
 
+    private cloneTransaction(transaction: Transaction): Transaction {
+        return Transaction.fromPlainObject(transaction.toPlainObject());
+    }
+
+    private cloneMessage(message: SignableMessage): SignableMessage {
+        return new SignableMessage({
+            message: message.message,
+            address: message.address,
+            signer: message.signer,
+            version: message.version
+        });
+    }
+
```

Comment:
Reordering to be read as a story is preferred (moving the private functions to the end is not desired).

E.g. https://stackoverflow.com/a/1760877/1475331